scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/shared/PortalConfig.java

```diff
@@ -0,0 +1,227 @@
+
+    /**
+     * Path to the user-defined portal property file
+     */
+    public static final String CONFIG_PATH = "scheduler-portal.conf";
+
+    /**
+     * Extra columns property
+     */
+    private static final String EXTRA_COLUMNS_PROPERTY = "execution-list-extra-columns";
+
+    /**
+     * Singleton behaviour
+     */
+    private static PortalConfig instance = null;
+
+    /**
+     * Config properties
+     */
+    private Map<String, String> properties = new HashMap<>();
+
+    /**
+     * Config properties
+     */
+    private List<JSONColumn> extraColumns = null;
+
+    /**
+     * @return current static config instance, cannot be null
+     */
+    public static PortalConfig get() {
+        if (instance == null) {
+            instance = new PortalConfig();
+        }
+        return instance;
+    }
+
+    public List<JSONColumn> getExtraColumns() {
+
+        if (extraColumns == null) {
+            String extraColumnsJson = properties.get(EXTRA_COLUMNS_PROPERTY);
+            List<JSONColumn> list = new ArrayList<>();
+            if (extraColumnsJson != null) {
+                JSONArray extraColumnsArray = null;
+                try {
+                    extraColumnsArray = JSONUtils.parseJSON(extraColumnsJson).isArray();
+                } catch (JSONException e) {
+                    LOGGER.log(Level.SEVERE, e.getMessage());
+                }
+
+                if (extraColumnsArray != null) {
+                    for (int i = 0; i < extraColumnsArray.size(); i++) {
+                        JSONObject extraColumnProperties = extraColumnsArray.get(i).isObject();
+
+                        if (extraColumnProperties != null) {
+                            JSONColumn column = new JSONColumn();
+                            column.setTitle(getTitle(extraColumnProperties, i));
+                            column.setName(getName(extraColumnProperties, i));
+                            column.setHidden(getHiddenDefault(extraColumnProperties, i));
+                            list.add(column);
+                        }
+                    }
+                }
+            }
+            extraColumns = list;
+        }
+        return extraColumns;
+    }
+
+    private String getTitle(JSONObject extraColumnProperties, int columnIndex) {
+        JSONString title = extraColumnProperties.get("title").isString();
+        if (title != null)
+            return title.stringValue();
+        else {
+            LOGGER.log(Level.SEVERE, "Could not read the field \"title\" of column number " + columnIndex);
+            return new String();
+        }
+    }
+
+    private String getName(JSONObject extraColumnProperties, int columnIndex) {
+
+        JSONObject information = extraColumnProperties.get("information").isObject();
+        if (information != null) {
+            String type = getType(information, columnIndex);
+            String key = getKey(information, columnIndex);
```

Comment:
As suggested before, please extract to a utility class. Same below in other parts.